### PR TITLE
YAML validation

### PR DESCRIPTION
**Pull request by sevos**
Early pull request, WIP, don't merge. Comments welcome

### CODE REVIEW

File: lib/rschema/aws_ami.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    private
+
+    def image(region, ami_id)
+      @images ||= {}
+      @images[region] ||= {}
+      @images[region][ami_id] ||= begin
+                                    ec2 = Aws::EC2::Client.new(region: region)
+                                    ec2.describe_images(image_ids: [ami_id],
+                                                        filters: [{name: 'state',
+                                                                   values: ['available']}]).
```
Comment (sevos) [inline lib/rschema/aws_ami.rb]: check for HVM AMI

File: lib/rschema/aws_region.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+module RSchema
+  module AwsRegion
+    REGIONS = %w(us-east-1 eu-west-1 eu-central-1)
+  end
+
+  module DSL
+    module Base
+      def in_each_region(regions = %w(us-east-1 eu-west-1 eu-central-1), &blk)
```
Comment (sevos) [inline lib/rschema/aws_region.rb]: reuse the constant

File: lib/rschema/aws_region.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+module RSchema
+  module AwsRegion
+    REGIONS = %w(us-east-1 eu-west-1 eu-central-1)
```
Comment (sevos) [inline lib/rschema/aws_region.rb]: list all regions

**Comment by suxor42** [comment]
Looks fine

**Pull request merged**